addons/web/static/src/webclient/actions/action_controller.js

```diff
@@ -0,0 +1,169 @@
+/** @odoo-module **/
+
+import { useBus } from "@web/core/bus_hook";
+import { cleanDomFromBootstrap } from "@web/legacy/utils";
+import { browser } from "@web/core/browser/browser";
+import { registry } from "@web/core/registry";
+
+const { Component, hooks, tags } = owl;
+const actionRegistry = registry.category("actions");
+
+let dialogCloseResolve;
```

Comment:
this is wrong, we may have multiple instances of actionService, hence relying on dialogCloseResolve to be a property within the component